File: frontend/src/services/report.service.js
import api from './api';

export const reportService = {
  getPasswordHealthReport: async () => {
    const response = await api.get('/reports/password-health');
    return response.data;
  },

  getLoginActivityReport: async () => {
    const response = await api.get('/reports/login-activity');
    return response.data;
  },
};

File: backend/src/main/java/com/securevault/service/ReportService.java
package com.securevault.service;

import com.securevault.dto.response.LoginActivityReportResponse;
import com.securevault.dto.response.PasswordHealthReportResponse;

public interface ReportService {

    PasswordHealthReportResponse getPasswordHealthReport(String userEmail);

    LoginActivityReportResponse getLoginActivityReport(String userEmail);
}
